### ISSUE TITLE
How to inference with pointcloud input?

### ISSUE THREAD

**Issue by qpc001** [facebookresearch/sam-3d-objects#20]

**Comment by gleize** [comment]
Hi @qpc001,

We do not provide the ability to use a pointcloud as input, however if you can convert your pointcloud to a pointmap, you can provide it [here](https://github.com/facebookresearch/sam-3d-objects/blob/cf066761706cd02b07e2fc6274570ec8cdafb683/notebook/inference.py#L106C9-L106C17) as argument.

**Comment by eternity123-null** [comment]
Hi @gleize , I have converted my depth into pointmap using 
`
def get_pointmap_pytorch(depth_tensor, K, depth_scale=1000.0):
    H, W = depth_tensor.shape
    device = depth_tensor.device
    
    fx = K[0, 0]
    fy = K[1, 1]
    cx = K[0, 2]
    cy = K[1, 2]

    y_range = torch.arange(H, device=device)
    x_range = torch.arange(W, device=device)
    u, v = torch.meshgrid(x_range, y_range, indexing='xy') 

    Z = depth_tensor / depth_scale
    X = (u - cx) * Z / fx
    Y = (v - cy) * Z / fy

    pointmap = torch.stack((X, Y, Z), dim=-1) 
    return pointmap
`
But it run into error like that:
<img width="3078" height="1028" alt="Image" src="https://github.com/user-attachments/assets/cc3e7b94-2aeb-42c3-9631-4012762cc810" />
Could you figure out what is going on?

**Comment by N0OBSTUDENT** [comment]
> Hi [@gleize](https://github.com/gleize) , I have converted my depth into pointmap using ` def get_pointmap_pytorch(depth_tensor, K, depth_scale=1000.0): H, W = depth_tensor.shape device = depth_tensor.device
> 
> ```
> fx = K[0, 0]
> fy = K[1, 1]
> cx = K[0, 2]
> cy = K[1, 2]
> 
> y_range = torch.arange(H, device=device)
> x_range = torch.arange(W, device=device)
> u, v = torch.meshgrid(x_range, y_range, indexing='xy') 
> 
> Z = depth_tensor / depth_scale
> X = (u - cx) * Z / fx
> Y = (v - cy) * Z / fy
> 
> pointmap = torch.stack((X, Y, Z), dim=-1) 
> return pointmap
> ```
> 
> ` But it run into error like that: <img alt="Image" width="2000" height="1028" src="https://private-user-images.githubusercontent.com/76623803/518094752-cc3e7b94-2aeb-42c3-9631-4012762cc810.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjQxNzY2NTMsIm5iZiI6MTc2NDE3NjM1MywicGF0aCI6Ii83NjYyMzgwMy81MTgwOTQ3NTItY2MzZTdiOTQtMmFlYi00MmMzLTk2MzEtNDAxMjc2MmNjODEwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTExMjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMTI2VDE2NTkxM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI0YWI4MWU4Y2NiNTFiZTM4YmMwZTliYWJkYmFlNmQzODk5MjJkZDBjNjQ5YWZiZTgwMTI2MDI4OTQ5NWRmZTkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.L8kAN4zew9s6AQS1cUbZQz10PKPEWHL-Mq1-CntG_4w"> Could you figure out what is going on?

@eternity123-null It's probably caused by zero/negative depth value. Replacing invalid depth with NaN works for me.

**Comment by eternity123-null** [comment]
Thanks for your helpful advise. I have solved this problem.

**Comment by qpc001** [comment]
> Thanks for your helpful advise. I have solved this problem.谢谢你的建议。我已经解决了这个问题。

Can you get the real metric of result?

**Comment by eternity123-null** [comment]
> > Thanks for your helpful advise. I have solved this problem.谢谢你的建议。我已经解决了这个问题。
> 
> Can you get the real metric of result?
Yes, the reason I metzero/negative depth value is that I didn't translate the coods into Pytorch3D frame.

**Comment by trungpham2606** [comment]
hi @eternity123-null 
you can get the real metric dimension of object by using pointmap ?
i also tried the pointmap of my object as suggested above but the generated mesh was still different in dimension.
i dont know whether i missed something ?

**Comment by trungpham2606** [comment]
hi @codepk37 
where did you add the point_map ? 
in this line, right?
https://github.com/facebookresearch/sam-3d-objects/blob/main/demo.py#L17

**Comment by codepk37** [comment]
> hi [@codepk37](https://github.com/codepk37) where did you add the point_map ? in this line, right? https://github.com/facebookresearch/sam-3d-objects/blob/main/demo.py#L17

Yes
